yt/data_objects/data_containers.py

```
@@ -662,6 +663,182 @@ def to_glue(self, fields, label="yt", data_collection=None):
+                `Firefly/data/yt`.
+
+            Returns
+            -------
+            reader : firefly_api.reader.Reader object
+                A reader object from the firefly_api, configured 
+                to output
+
+            Examples
+            --------
+
+                >>> ramses_ds = yt.load(
+                ...     "/Users/agurvich/Desktop/yt_workshop/"+
+                ...     "DICEGalaxyDisk_nonCosmological/output_00002/info_00002.txt")
+
+                >>> region = ramses_ds.sphere(ramses_ds.domain_center,(1000,'kpc'))
+
+                >>> reader = region.create_firefly_object(
+                ...     path_to_firefly="/Users/agurvich/research/repos/Firefly",
+                ...     fields_to_include=[
+                ...     'particle_extra_field_1',
+                ...     'particle_extra_field_2'],
+                ...     fields_units = ['dimensionless','dimensionless'],
+                ...     dataset_name = 'IsoGalaxyRamses')
+
+                >>> reader.options['color']['io']=[1,1,0,1]
+                >>> reader.particleGroups[0].decimation_factor=100
+                >>> reader.dumpToJSON()
+            """ 
+        
+        ## attempt to import firefly_api
+        try:
+            from firefly_api.reader import Reader
+            from firefly_api.particlegroup import ParticleGroup
+            from firefly_api.errors import FireflyError,FireflyWarning
+            from firefly_api.errors import warnings as firefly_warnings
+        except ImportError:
+            raise ImportError("Can't find firefly_api, ensure it"+
+                "is in your python path or install it with"+
+                "'$ pip install firefly_api'. It is also available"+
+                "on github at github.com/agurvich/firefly_api")
+        try:
+            assert len(fields_units) == len(fields_to_include)
+        except AssertionError:
+            raise FireflyError("Each requested field must have units.")
```

I would phrase this try/except like this:

```
if len(field_units) != len(fields_to_include):
    raise RuntimeError("Each requested field must have units.")
```

Rather than catching the assertion you can just use an if statement and raise whatever error you want to raise if the if statement is not true. It's also  a little weird to raise an error from an outside package from inside yt, so rather than importing errors and warnings from firefly, just use either regular python errors and warnings, or if you really need custom errors, then use the ones from yt.

---

Sure thing.